DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -961,6 +964,44 @@ public async Task<byte[]> HandleHttpInteractionAsync(ArraySegment<byte> body, Ca
         return await interaction.GetResponseAsync();
     }
 
+    [StackTraceHidden]
+    internal ChannelReader<GuildMembersChunkedEventArgs> RegisterGuildMemberChunksEnumerator(ulong guildId, string? nonce)
+    {
+        Int128 nonceKey = new(guildId, (ulong)(nonce?.GetHashCode() ?? 0));
+        Channel<GuildMembersChunkedEventArgs> channel = Channel.CreateUnbounded<GuildMembersChunkedEventArgs>(new UnboundedChannelOptions { SingleWriter = true, SingleReader = true });
+
+        if (!this.guildMembersChunkedEvents.TryAdd(nonceKey, channel))
+        {
+            throw new InvalidOperationException("A guild member chunk request for the given guild and nonce has already been registered.");
+        }
+
+        return channel.Reader;
+    }
+
+    private async ValueTask DispatchGuildMembersChunkForIteratorsAsync(GuildMembersChunkedEventArgs eventArgs)
+    {
+        if (this.guildMembersChunkedEvents.Count is 0)
+        {
+            return;
+        }
+
+        Int128 code = new(eventArgs.Guild.Id, (ulong)(eventArgs.Nonce?.GetHashCode() ?? 0));
+
+        if (!this.guildMembersChunkedEvents.TryGetValue(code, out Channel<GuildMembersChunkedEventArgs>? eventChannel))
+        {
+            return;
+        }
+
+        await eventChannel.Writer.WriteAsync(eventArgs);
+
+        // Discord docs state that 0 <= chunk_index < chunk_count, so add one

```

Comment:
So its basically 0 indexed? That coment confused me

Reply:
From what i understand